src/iceberg/avro/avro_schema_util.cc

```diff
@@ -188,8 +213,19 @@ Status ToAvroNodeVisitor::Visit(const StructType& type, ::avro::NodePtr* node) {
     ::avro::NodePtr field_node;
     ICEBERG_RETURN_UNEXPECTED(Visit(sub_field, &field_node));
 
-    // TODO(gangwu): sanitize field name
-    (*node)->addName(std::string(sub_field.name()));
+    // Sanitize field name to ensure it follows Avro field name requirements
+    std::string sanitized_name = SanitizeFieldName(sub_field.name());
```

Comment:
In most cases, field names do not need sanitization. This may introduce a large regression.